pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -462,5 +462,9 @@ func (m *manager) updateContainerCPUSet(containerID string, cpus cpuset.CPUSet)
 }
 
 func (m *manager) GetCPUs(podUID, containerName string) cpuset.CPUSet {
-	return m.state.GetCPUSetOrDefault(podUID, containerName)
+	if result, ok := m.state.GetCPUSet(string(podUID), containerName); ok {
```

Comment:
We should add podresource API end to end tests to explain and emphasize the rationale behind this change.

Reply:
Codewise I see no issue with this change. The question is: what is a "concrete resource assignment" for CPUs? this is what concerns podresources API. I never found a very strong definition, so the best shot is trying to understand the original intent of the API, [which was meant to track devices](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/606-compute-device-assignment#user-stories).
So, what is a "cpu assigment" for a container?
In the current form, the API is telling its consumers on which CPUs a container is allowed to run. This tells us little, because we can't distinguish easily exclusive allocated cpus or not.
With this change, we report only the exclusively allocated cpus, which is
1. closer to the original API semantics as I understand it and
2.  gives MORE information to the caller:
    a. if a container has exclusively allocated CPUs, well, it can run only on those cpus. All the information is available.
   b. if a container has NOT exclusively allocated CPUs, it can run on all the allocatable-non-exclusively-allocated CPUs, but this set can constructed using the information provided by the podresources API: get allocatable cpus from `GetAllocatable`, then remove all the exclusive sets we return now. This makes the API self-sufficient and in a better shape than it is today.

For all these reasons, I think this is a good change and I'm fine with this direction.